SNIPS/snip-18.md

```diff
@@ -0,0 +1,159 @@
+
+This SNIP aims for the first stage of staking to be live on Starknet mainnet in Q4 2024, featuring a permissionless on-chain staking protocol and stake delegation.
+
+In this SNIP, we will describe in detail the staking proposal, its timeline, and its relation to future steps in decentralization.
+
+## Motivation
+
+The first stage of staking marks another crucial step in Starknet’s decentralized [vision](https://starkware.co/integrity-matters-blog/starknet-vision/). Here are the key motivations for an incremental approach in Starknet’s decentralization process and for the proposed first step.
+
+In the future, as part of Starknet’s Consensus protocol, Stakers will be responsible for maintaining and securing the network by producing, attesting, and proving blocks. However, it’s not feasible to hand over these responsibilities all in one day. This is why an incremental approach is necessary, as it allows for the following:
+
+* **Gradual Implementation:** A PoS protocol is a complex structure that requires comprehensive testing and validation. By introducing its features in small, manageable milestones, we ensure that each step is carefully implemented and easily integrated. This approach also allows us to refine the process and address issues as they arise.
+* **Stakers' Preparation:** Giving Stakers time to adapt to new responsibilities, ensuring they are ready for their critical roles, and maximizing network stability during the transition.
+* **Proven Reliability:** By the time Stakers perform the aforementioned roles, we aim to have a proven set of sequencers that reliably produce and attest blocks, ensuring the network's stability and performance.
+
+Specifically, in this proposed first stage, we will test the protocol’s economic incentive structure and some of its key smart contract components. The protocol reward structure must balance incentivizing participation with maintaining a sustainable inflation rate and allowing a sufficient amount of STRK to remain readily available for other network activity.
+
+By adopting this step-by-step approach, with the proposed first stage, we can gather valuable data, perform crucial tests, and thus refine the staking protocol as we implement it.
+
+## Specification
+
+### Overview
+
+The proposed first stage protocol features permissionless staking of STRK on Starknet with two staking flavors:
+
+* **Staking:** As a Staker, you can stake any amount of STRK greater than X STRK (X is between 10K-100K). Currently, Stakers are expected to run full nodes in preparation for the following stages. In the future, they will also need to run additional software that sequences and validates Starknet blocks and possibly perform additional network liveness and security tasks.
```

Comment:
Is there an upper limit of how many Stakers you expect/would like to have?
Or is it going to be like on Atom where the X Stakers with most STRK will be receiving rewards?

Reply:
We don't plan to have a limit or have the Atom mechanism. Instead, if the number of Stakers exceeds 100, we will choose a committee of 100 randomly (with the probability to be chosen proportional to the staking power) for each epoch.